### PR TITLE
feat: Bids proposal view

### LINKED ISSUE #1103: Bids proposal view
Bids proposal view should have the same structure as grants proposal view

### PR DESCRIPTION
**Pull request by ncomerci**
This closes #1103

### CODE REVIEW

File: src/pages/proposal.tsx [hunk 1]
```diff
@@ -123,6 +124,17 @@ function formatDescription(description: string) {
   return position > 0 ? value.slice(0, position).trim() : value
 }
 
+function getProposalView(proposal: ProposalAttributes | null) {
+  switch (proposal?.type) {
```
Comment (andyesp) [inline src/pages/proposal.tsx:128]: Maybe return null if proposal is null?

Comment (ncomerci) [reply src/pages/proposal.tsx:128]: In that case, I think a "null" text would be displayed

File: src/entities/Bid/routes.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+export default routes((route) => {
+  const withAuth = auth()
+  route.get('/bids/:tenderId/has-user-placed-bid', withAuth, handleAPI(hasUserPlacedBid))
+  route.get('/bids/:tenderId/is-submission-finished', handleAPI(isSubmissionFinished))
+})
+async function hasUserPlacedBid(req: WithAuth) {
+  const user = req.auth!
+  const { tenderId } = req.params
+  return await BidService.hasUserBidOnTender(user, tenderId)
```
Comment (andyesp) [inline src/entities/Bid/routes.ts:16]: If this is returning an object with some bid data could we change its name? Maybe `getUserBid`?

File: src/pages/submit/bid.tsx [hunk 3]
```diff
@@ -106,6 +110,14 @@ export default function SubmitBid() {
     pitchProposal?.snapshot_id || '',
     pitchProposal?.configuration.choices
   )
+  const hasUserPlacedBid = useGetPlacedBid(linkedProposalId)
```
[the remark this answers is not in the document]
Comment (1emu) [reply src/pages/submit/bid.tsx:113]: `useGetPlacedBid` could be `useGetUserBid`, to explicit that it's not any placed bid

**Review by 1emu — APPROVED** [review]

File: src/hooks/useGetUserBid.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { useQuery } from '@tanstack/react-query'
+
+import { Governance } from '../clients/Governance'
+
+function useGetUserBid(tenderId: string | null) {
```
Comment (andyesp) [inline src/hooks/useGetUserBid.ts:5]: `useUserBids`

**Review by 1emu — APPROVED** [review]

**Review by andyesp — APPROVED** [review]

**Pull request merged**